public/js/tracking_protection.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-env browser */
+"use strict";
+
+/* 
+	Respect user privacy settings and honor user privacy choices.
+	From Schalk Neethling at https://github.com/schalkneethling/dnt-helper */
+	
+const trackingProtection = {};
+trackingProtection.enabled = function(dnt, ua){
+	window.dataLayer = window.dataLayer || [];
+	let userPrivacySettings = navigator.doNotTrack || window.doNotTrack || navigator.msDoNotTrack;
+	const userAgent = ua || navigator.userAgent;
+	const anomalousWinVersions = ["Windows NT 6.1", "Windows NT 6.2", "Windows NT 6.3"];
+	const fxMatch = userAgent.match(/Firefox\/(\d+)/);
+	const ieRegEx = /MSIE|Trident/i;
+	const isIE = ieRegEx.test(userAgent);
+	const platform = userAgent.match(/Windows.+?(?=;)/g);
+	if(isIE&&typeof Array.prototype.indexOf !== "function") {
+		return false;
+	} 
+	else if(fxMatch&&parseInt(fxMatch[1],10)<32) {
+		userPrivacySettings="Unspecified";
+	} 
+	else if(isIE && platform&&anomalousWinVersions.indexOf(platform.toString()) !== -1) {
+		userPrivacySettings="Unspecified";
+	}
+	else {
+		userPrivacySettings = {
+		0 : "Disabled",
+		1 : "Enabled",
+		}[userPrivacySettings] || "Unspecified";}
```

Comment:
I had to read this a few times to try and understand the temp object and array index code here. Not sure if there is a clearer way to refactor this for legibility (since userPrivacySettings can be an int or string or already Unspecified.

Reply:
Actually, why bother with Unspecified at all since we only ever check if it equals Enabled below?

Reply:
Ah, missed the sneaky trailing `}`, which I believe closes the else statement. We may want to try running this through a formatter or something to ensure consistent indenting, etc. ... or leave it for now, and I can try running codebase through prettier.

Reply:
I thought my powers of copy/paste/rename vars were stronger than this but clearly some things got lost in transit from Schalk Neethling's most excellent dnt-helper repo to my branch. I would in any case like to go through this later and make it more "regular human readable" in the hopes that passive javascripters might be less intimated/more inclined to use it.